pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -309,11 +316,19 @@ func (tm *TableManager) ensureQueryReadiness(ctx context.Context) error {
 			return err
 		}
 
+		for _, u := range usersToBeQueryReadyFor {
+			distinctUsers[u] = struct{}{}
+		}
+
+		perTableStart := time.Now()
 		if err := table.EnsureQueryReadiness(ctx, usersToBeQueryReadyFor); err != nil {
 			return err
 		}
+		joinedUsers := strings.Join(usersToBeQueryReadyFor, ",")
+		level.Debug(util_log.Logger).Log("msg", "instance query ready for users", "users", joinedUsers, "query_readiness_duration", time.Since(perTableStart), "table", tableName)
```

Comment:
The log here would be misleading since the `distinctUsers` is used for all the tables and it is possible that not all the users would have an index in all the tables, for e.g some users stopping to push logs. Also, the list could get pretty huge so let us just log a count here and the next line?